runtime/centrifuge/src/lib.rs

```diff
@@ -89,6 +93,78 @@ pub fn native_version() -> NativeVersion {
 	}
 }
 
+/// Custom runtime upgrades
+///
+/// Migration to include collator-selection in a running chain
+
+pub struct IntegrateCollatorSelection<T>(PhantomData<T>);
+
+type BalanceOfCollatorSelection<T> =
+	<<T as pallet_collator_selection::Config>::Currency as Currency<
+		<T as frame_system::Config>::AccountId,
+	>>::Balance;
+
+impl<T> IntegrateCollatorSelection<T>
+where
+	T: pallet_session::Config + pallet_collator_selection::Config + frame_system::Config,
+	T::AccountId: From<sp_runtime::AccountId32>,
+	T::Keys: From<SessionKeys>,
+{
+	fn to_version() -> u32 {
+		1010
+	}
+
+	fn db_access_weights(reads: Option<u64>, writes: Option<u64>) -> Weight {
+		let mut weight: Weight = 0u32 as Weight;
+
+		if let Some(num_reads) = reads {
+			weight += T::DbWeight::get()
+				.reads(1 as Weight)
+				.saturating_mul(num_reads);
+		}
+
+		if let Some(num_writes) = writes {
+			weight += T::DbWeight::get()
+				.writes(1 as Weight)
+				.saturating_mul(num_writes);
+		}
+
+		weight
+	}
+}
+
+impl<T> OnRuntimeUpgrade for IntegrateCollatorSelection<T>
+where
+	T: pallet_session::Config + pallet_collator_selection::Config + frame_system::Config,
+	BalanceOfCollatorSelection<T>: From<u128>,
+	T::AccountId: From<sp_runtime::AccountId32>,
+	T::Keys: From<SessionKeys>,
+	<T as frame_system::Config>::AccountId: From<<T as pallet_session::Config>::ValidatorId>,
+{
+	fn on_runtime_upgrade() -> Weight {
+		let mut consumed: Weight = 0;
+
+		if VERSION.spec_version == IntegrateCollatorSelection::<T>::to_version() {
+			let mut current_validators = <pallet_session::Validators<T>>::get()
+				.into_iter()
+				.map(|who| who.into())
+				.collect::<Vec<_>>();
+
+			current_validators.truncate(T::MaxInvulnerables::get() as usize);
+
+			// SAFETY: The vector is truncate before to ensure the vector fit into the
+			// expected BoundedVec from pallet_collator_selection.
+			let current_validators = current_validators.try_into().unwrap();
```

Comment:
I needed to add this section in addition to how it was on [Altair](https://github.com/centrifuge/centrifuge-chain/blob/v3.0.0-alpha7-2/runtime/altair/src/lib.rs#L94-L154) because now it is working with `Boundedvec`. Also the limit of immutable validators is 100. I hope it is enough

Reply:
we only have 3, so no problem since it will not be executed later in time.